Lib/urllib/request.py

```diff
@@ -885,8 +885,15 @@ class AbstractBasicAuthHandler:
 
     # allow for double- and single-quoted realm values
     # (single quotes are a violation of the RFC, but appear in the wild)
-    rx = re.compile('(?:.*,)*[ \t]*([^ \t]+)[ \t]+'
-                    'realm=(["\']?)([^"\']*)\\2', re.I)
+    rx = re.compile('(?:^|,)'   # start of the string or ','
+                    '[ \t]*'    # optional whitespaces
+                    '([^ \t]+)' # scheme like "Basic"
+                    '[ \t]+'    # mandatory whitespaces
+                    # realm=xxx
```

Comment:
Could you remove this, please?

Reply:
I'm sorry, I don't understand your request.

This PR is a straightforward backport of https://github.com/python/cpython/commit/0b297d4ff1c0e4480ad33acae793fbaf4bf015b4 The code in master also contains the comments like "realm=xxx".

The following comment shows which formats are accepted for the realm field:

```
# realm=xxx
# realm='xxx'
# realm="xxx"
```

See unit tests for more concrete examples.

If you want to remove/rephrase the comments, it should be done first in the master branch.

Reply:
Oh, okay.  I thought this was debug test code that got left in, like leaving a scalpel in the patient. ;-)